interfaces/cython/cantera/test/test_composite.py

```diff
@@ -145,6 +145,53 @@ def test_collect_data(self):
         self.assertIn('X', collected)
         self.assertEqual(collected['X'].shape, (0, self.gas.n_species))
 
+    def test_append_no_norm_data(self):
+        gas = ct.Solution("h2o2.yaml")
+        gas.TP = 300, ct.one_atm
+        gas.X = gas.X - 1.e-16
+        states = ct.SolutionArray(gas)
+        states.append(T=gas.T, P=gas.P, X = gas.X, normalize = False)
+        self.assertEqual(states[0].T, gas.T)
+        self.assertEqual(states[0].P, gas.P)
+        self.assertArrayNear(states[0].X, gas.X)
+
+        w = ct.Water()
```

Comment:
This should be a new test, IMO, to limit the number of assertions per test. It's usually best to keep the scope of the test to one class or function or behavior.